openlibrary/plugins/openlibrary/js/initAnnouncementBanner.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Init the announcement banners if any
+ * If patron has dismissed the banner by clicking the 'X' button, banner storage key will be added to localstorage
+ * Banners with storage keys within the localstorage will not show again
+ */
+export function initAnnouncementBanner() {
+    const announcementBanner = document.getElementById('announcement-banner');
+    const storedHiddenBannerKeys = localStorage.getItem('hiddenBannerKeys');
```

Comment:
There shouldn't be a `hiddenBannerKeys` object in local storage.  Use the banner key as the local storage key.